notebooks/tests/test2_tellseq_C_equal_volume_pooling.py

```diff
@@ -0,0 +1,159 @@
+        for sub in EXPECTED_SUBDIRS:
+            candidate = os.path.join(self.repo_test_output_dir, sub, filename)
+            if os.path.exists(candidate):
+                return candidate
+        raise AssertionError(
+            f"[COMPARE] Expected golden file not found for '{filename}'. "
+            f"Searched under: {', '.join(EXPECTED_SUBDIRS)}"
+        )
+
+    def _assert_text_files_equal(self, produced_fp: Path, expected_fp: Path):
+        """Strict line-by-line text comparison with helpful diffs."""
+        with open(produced_fp, "r") as f_out, open(expected_fp, "r") as f_exp:
+            out_lines = f_out.readlines()
+            exp_lines = f_exp.readlines()
+
+        self.assertEqual(
+            len(out_lines), len(exp_lines),
+            msg=(f"[DIFF] Line count differs for '{produced_fp.name}': "
+                 f"produced={len(out_lines)} vs expected={len(exp_lines)}")
+        )
+
+        for i, (o, e) in enumerate(zip(out_lines, exp_lines), start=1):
+            if o != e:
+                self.fail(
+                    f"[DIFF] Mismatch in '{produced_fp.name}' at line {i}.\n"
+                    f"Produced: {o.rstrip()}\n"
+                    f"Expected: {e.rstrip()}\n"
+                )
+
+    def test_equal_volume_pooling_outputs_shadow(self):
+        """
+        Keep Notebook C's hardcoded './test_output/...' as-is by creating that
+        structure under a temp dir and running with cwd=tmp. Then compare tmp outputs
+        vs repo goldens by filename.
+        """
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tmp_root = Path(tmp_dir)            # shadow root = './'
+            tmp_out  = tmp_root / "test_output" # ./test_output
+            tmp_qc   = tmp_out / "QC"
+            tmp_idx  = tmp_out / "Indices"
+            tmp_ss   = tmp_out / "SampleSheets"
+
+            # create expected dirs under tmp
+            for d in (tmp_qc, tmp_idx, tmp_ss):
+                d.mkdir(parents=True, exist_ok=True)
+
+            # copy required inputs from repo -> tmp
+            shutil.copy2(
+                os.path.join(self.repo_test_output_dir, "QC", "Tellseq_plate_df_B.txt"),
+                tmp_qc / "Tellseq_plate_df_B.txt",
+            )
+            shutil.copy2(
+                os.path.join(self.repo_test_output_dir, "QC", "Tellseq_expt_info.yml"),
+                tmp_qc / "Tellseq_expt_info.yml",
+            )
+     
+           
+            run_params = {
+                "current_set_id": "col19to24",
+                "total_vol": 190,
+                "iseq_sequencer": "iSeq",
+                "novaseq_sequencer": "NovaSeqXPlus",
+                "full_plate_fp": "./test_output/QC/Tellseq_plate_df_B.txt",
+                "expt_config_fp": "./test_output/QC/Tellseq_expt_info.yml",
+            }
+    
+            # run notebook with cwd switched to tmp_root so './test_output' points to tmp
+            prev_cwd = os.getcwd()
```

Comment:
I do not think this is the direction we want to go ... changing the working directory is a lot more work (and more error prone) than simply changing the input path strings!